libyul/AST.h

```diff
@@ -59,7 +60,7 @@ struct VariableDeclaration { langutil::DebugData::ConstPtr debugData; TypedNameL
 /// Block that creates a scope (frees declared stack variables)
 struct Block { langutil::DebugData::ConstPtr debugData; std::vector<Statement> statements; };
 /// Function definition ("function f(a, b) -> (d, e) { ... }")
-struct FunctionDefinition { langutil::DebugData::ConstPtr debugData; YulString name; TypedNameList parameters; TypedNameList returnVariables; Block body; };
+struct FunctionDefinition { langutil::DebugData::ConstPtr debugData; YulName name; TypedNameList parameters; TypedNameList returnVariables; Block body; };
```

Comment:
We may have a small problem here. With this change, user-defined functions will use `YulName`, while the built-in functions in the dialect will still be identified with `YulString`. Since the latter by design have fixed names, I'm not sure it's even feasible to use `YulName` for them. But then `FunctionCall` node will not be able to refer to them.